src/components/BlogPostViewer.tsx

```diff
@@ -78,33 +81,19 @@ export const BlogPostViewer: React.FC<BlogPostViewerProps> = ({
               comments,
               user,
             }) => (
-              <>
-                <div key={id} className="mb-6">
-                  <BlogPostComponent
-                    id={id}
-                    authorId={userId}
-                    title={title}
-                    lastUpdated={updatedAt}
-                    createdAt={createdAt}
-                    content={content}
-                    comments={(comments as Comment[]).length}
-                    authorName={user.name as string}
-                    imgUrl={user.image}
-                  />
-                </div>
-                <CommentModal
```

Comment:
It might've been hard to notice with few blog posts but this actually renders one invisible modal for each blog post on the blog post feed. I've refactored CommentModal to only render when there is a selected post (ie. when opening one from the blog post viewer or from the search bar). Rendering one comment modal for each blog posts was causing some pretty heavy performance issues throughout the site - should feel noticably faster now.